core/primitives/src/delegate_action.rs

```diff
@@ -0,0 +1,251 @@
+//! DelegateAction is a type of action to support meta transactions.
+//!
+//! NEP: https://github.com/near/NEPs/pull/366
+//! This is the module containing the types introduced for delegate actions.
+
+pub use self::private_non_delegate_action::NonDelegateAction;
+use crate::hash::{hash, CryptoHash};
+use crate::transaction::Action;
+use crate::types::{AccountId, Nonce};
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_crypto::{PublicKey, Signature};
+use near_primitives_core::types::BlockHeight;
+use serde::{Deserialize, Serialize};
+use std::io::{Error, ErrorKind};
+
+/// This is an index number of Action::Delegate in Action enumeration
+const ACTION_DELEGATE_NUMBER: u8 = 8;
+/// This action allows to execute the inner actions behalf of the defined sender.
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Eq, Clone, Debug)]
+pub struct DelegateAction {
+    /// Signer of the delegated actions
+    pub sender_id: AccountId,
+    /// Receiver of the delegated actions.
+    pub receiver_id: AccountId,
+    /// List of actions to be executed.
```

Comment:
nit: I'd also mention here, that we don't allow the 'nested' DelegateActions (that's why there is a NonDelegateAction entry there)

Reply:
done